tests/test_timer/timer.py

```diff
@@ -0,0 +1,69 @@
+import logging
+import time
+from contextlib import contextmanager
+from threading import RLock
+
+import numpy
+
+__all__ = ["Timer"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class Timer:
+    """Timer to log timings during inference requests. Should be used through
+    timer_utils.get_singleton_manager()
+    """
+
+    _instance = None
+
+    def __init__(self, enable_logging: bool):
+        """
+        :param enable_logging: whether or not time logging is enabled
+        """
+        self.enable_logging = enable_logging
+        self.measurements = {}
```

Comment:
nit: We could make it a `defaultdict`, but we can ignore since these are tests

Reply:
This will be helpful to instantiate the default diict with a list for appending the tim elapsed.